interactive_engine/executor/ir/graph_proxy/src/apis/partitioner.rs

```diff
@@ -16,13 +16,17 @@
 use crate::apis::ID;
 use crate::GraphProxyResult;
 
-pub trait Partitioner: Send + Sync + 'static {
-    /// Given the element id and job_workers (number of worker per server),
-    /// return the id of worker that is going to process
-    fn get_partition(&self, id: &ID, job_workers: usize) -> GraphProxyResult<u64>;
-    /// Given job_workers (number of worker per server) and worker_id (worker index),
-    /// return the partition list that the worker is going to process
-    fn get_worker_partitions(
-        &self, job_workers: usize, worker_id: u32,
-    ) -> GraphProxyResult<Option<Vec<u64>>>;
+pub type PartitionId = u32;
+pub type ServerId = u32;
+
+/// A `PartitionInfo` is used to query the partition information when the data has been partitioned.
+pub trait PartitionInfo: Send + Sync + 'static {
+    /// Given the data, return the id of the partition that holds the data.
+    fn get_partition_id(&self, data: &ID) -> GraphProxyResult<PartitionId>;
```

Comment:
ID for data is not very nice. We may want to define `trait PartitionedData`, even we may not need any function for them.
```
pub trait PartitionedData {  }

impl PartitionedData for Vertex {  }
impl PartitionedData for Edge {  }

pub trait PartitionInfo: Send + Sync + 'static {
    fn get_partition_id<D: PartitionedData>(&self, data: &Data) -> GraphProxyResult<PartitionId>;
}

```